include/cuco/detail/open_addressing/open_addressing_ref_impl.cuh

```diff
@@ -595,12 +595,12 @@ class open_addressing_ref_impl {
           case insert_result::DUPLICATE: return false;
           default: continue;
         }
-      } else if (group.any(state == detail::equal_result::EMPTY)) {
-        // Key doesn't exist, return false
-        return false;
-      } else {
-        ++probing_iter;
       }
+
+      // Key doesn't exist, return false
+      if (group.any(state == detail::equal_result::EMPTY)) { return false; }
+
+      ++probing_iter;
```

Comment:
I still don't get why the old logic was wrong but if it fixes the problem with the flaky unit test then I'm down with it.

Reply:
Me neither. If this doesn't fix the issue, at least the new test increases the chance of test failures thus we can take a closer look at the point.